### PR TITLE
[WPB-15151] eliminate legahold test redundancy

### PR DESCRIPTION
**Pull request by fisx**
## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Comment by fisx** [comment]
commits not in #4384:

c050b30
26f8961
f90ee92

File: integration/test/Test/LegalHold.hs [hunk 1]
```diff
@@ -1102,3 +1107,26 @@ testNoCommonVersion = do
     bindResponse (requestLegalHoldDevice tid alice bob) $ \resp -> do
       resp.status `shouldMatchInt` 500
       resp.json %. "label" `shouldMatch` "server-error"
+
+-- | LH can be configured i a way that does not require users to give preliminary consent to
```
Comment (supersven) [inline integration/test/Test/LegalHold.hs:1111]:
```suggestion
-- | LH can be configured in a way that does not require users to give preliminary consent to
```

File: integration/test/Test/LegalHold.hs [hunk 2]
```diff
@@ -1102,3 +1107,26 @@ testNoCommonVersion = do
...
+-- LH when being added to a team.  The user still has to approve the LH device before the
+-- recording starts.  This is called "implicit consent", was introduced to accomodate specific
+-- work flows, and there is some hope that it'll be removed in the future.
+--
+-- Explicit consent requires users to consent on entering the team, and then approve the
+-- actual being put under recording again if it happens.
+--
+-- This flag allows to make tests run through both configurations with minimal adjustment.
+data ImplicitConsent = ImplicitConsent | ExplicitConsent
+  deriving (Eq, Show, Generic)
```
Comment (supersven) [inline integration/test/Test/LegalHold.hs:1121]: Where does this need `Generic`? :thinking:

File: integration/test/Test/LegalHold.hs [hunk 3]
```diff
@@ -1102,3 +1107,26 @@ testNoCommonVersion = do
...
+
+setLHFeatureConfigForServer :: ImplicitConsent -> (String {- domain -} -> App ()) -> App ()
+setLHFeatureConfigForServer ImplicitConsent app =
+  -- we could do `setField "settings.featureFlags.legalhold"
+  -- "whitelist-teams-and-implicit-consent"`, but this is already the default.
+  -- TODO: make this an assertion, not a comment!
```
Comment (supersven) [inline integration/test/Test/LegalHold.hs:1127]: Can't this easily be done now?

**Review by supersven — CHANGES_REQUESTED** [review]

File: integration/test/Test/LegalHold.hs (hunk 2 above)
Comment (fisx) [reply integration/test/Test/LegalHold.hs:1121]: for the test collection machinery

File: integration/test/Test/LegalHold.hs (hunk 3 above)
Comment (fisx) [reply integration/test/Test/LegalHold.hs:1127]: right.  missed that spot, thanks!

**Comment by fisx** [comment]
thanks, sven!

the CI failures seem unrelated?

**Comment by supersven** [comment]
> thanks, sven!
> 
> the CI failures in [64193e7](https://github.com/wireapp/wire-server/pull/4386/commits/64193e79b904121608437839b93aae37b70b0d47) seem unrelated?

More :snowflake: :grimacing:

**Review by supersven — APPROVED** [review]
IMHO we can merge this as soon as the CI is :green_circle:

**Pull request merged**